op-node/rollup/derive/channel_test.go

```diff
@@ -99,3 +106,60 @@ func TestFrameValidity(t *testing.T) {
 		t.Run(tc.name, tc.Run)
 	}
 }
+
+func TestBatchReader(t *testing.T) {
+	// Get batch data
+	rng := rand.New(rand.NewSource(0x543331))
+	singularBatch := RandomSingularBatch(rng, 5000, big.NewInt(333))
+	batchDataInput := NewBatchData(singularBatch)
+
+	encodedBatch := &bytes.Buffer{}
+	// Get the encoded data of the batch data
+	batchDataInput.encodeTyped(encodedBatch)
+
+	var testCases = []struct {
+		name string
+		algo func(buf *bytes.Buffer)
+	}{{
+		name: "zlib",
+		algo: func(buf *bytes.Buffer) {
+			writer := zlib.NewWriter(buf)
+			writer.Write(encodedBatch.Bytes())
+		},
+	},
+		{
+			name: "brotli10",
+			algo: func(buf *bytes.Buffer) {
+				buf.WriteByte(ChannelVersionBrotli)
+				writer := brotli.NewWriterLevel(buf, 10)
+				writer.Write(encodedBatch.Bytes())
+			},
+		}, {
```

Comment:
This round trip test is important, _here_ we should test all comression algos, not only Brotli10 :) Would be good to also add 9 and 11.